modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -217,6 +251,10 @@ def _read_rows(
         quotechar: bytes = b'"',
         is_quoting: bool = True,
         outside_quotes: bool = True,
+        max_bytes: int = None,
```

Comment:
I spent some time in past for splitting work with bytes and rows: https://github.com/modin-project/modin/pull/2344.

Please don't mix again.

Reply:
@anmyachev `max_bytes` was added here to create `stop_condition` for `_read_rows` function when `nrows` is None, but `skiprows` is not None and should be processed by `_read_rows` function (in that case we have to process data file in `row-by-row` manner because of `skiprows`). If we remove `max_bytes` from `_read_rows` function, we still need to define some criteria to define partition size. In the last commit d2bf617a3065bb358b8fd639de3441107dcf9ad0 i have implemented such criteria via `lines_amount` parameter (number of lines in the data file), comparison of this and previous implementations:
```
       before           after         ratio

     <issue-1067~1>       <issue-1067>
       2.82±0.01s       2.98±0.03s     1.06  io.csv.TimeReadCsvSkiprows.time_skiprows((5000, 5000), 'lambda_even_rows')
       2.82±0.03s       2.98±0.03s     1.06  io.csv.TimeReadCsvSkiprows.time_skiprows((5000, 5000), 'range_step2')
          5.46±0s       8.19±0.07s    ~1.50  io.csv.TimeReadCsvSkiprows.time_skiprows((50000, 5000), 'lambda_even_rows')
       5.67±0.02s       8.38±0.03s    ~1.48  io.csv.TimeReadCsvSkiprows.time_skiprows((50000, 5000), 'range_step2')
```
Measurements show performance degradation of the new implementation especially with increased number of rows (`lines_amount` can be optimized, but it will add overhead anyway).

Reply:
Let's get back to the approach with `max_bytes`

Reply:
Approach was returned.